File: 백준/Silver/18870. 좌표 압축/좌표 압축.py
import sys


N = int(sys.stdin.readline())
line_words = list(map(int, sys.stdin.readline().rstrip().split(" ")))
sorted_word = sorted(list(set(line_words)))

dic = {sorted_word[i] :i for i in range(len(sorted_word))}

for i in line_words:
    print(dic[i], end=" ")

File: 백준/Silver/18870. 좌표 압축/좌표 압축.java
import java.io.*;
import java.util.*;
import java.util.stream.Collectors;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br= new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int tryNum = Integer.parseInt(br.readLine());
        List<Integer> points = Arrays.stream(br.readLine().split(" "))
                .map(Integer::parseInt)
                .collect(Collectors.toList());

        List<Integer> sorted = points.stream()
                .sorted()
                .distinct()
                .collect(Collectors.toList());

        Map<Integer, Integer> set = new HashMap<>();

        for(int node:sorted){
            set.put(node, set.size());
        }

        for(int i=0;i<tryNum;i++){
            bw.write(set.get(points.get(i))+ " ");
        }
        bw.flush();

    }
}
